sympy/assumptions/sathandlers.py

```diff
@@ -220,62 +241,28 @@ def __call__(self, expr):
 
 ## Abs ##
 
-@class_fact_registry.register(Abs)
-def _(expr):
-    return Q.nonnegative(expr)
-
-@class_fact_registry.register(Abs)
-def _(expr):
-    arg = expr.args[0]
-    return Equivalent(~Q.zero(arg), ~Q.zero(expr))
-
 @class_fact_registry.register(Abs)
 def _(expr):
     arg = expr.args[0]
-    return Implies(Q.even(arg), Q.even(expr))
-
-@class_fact_registry.register(Abs)
-def _(expr):
-    arg = expr.args[0]
-    return Implies(Q.odd(arg), Q.odd(expr))
-
-@class_fact_registry.register(Abs)
-def _(expr):
-    arg = expr.args[0]
-    return Implies(Q.integer(arg), Q.integer(expr))
+    return [Q.nonnegative(expr), 
+            Equivalent(~Q.zero(arg), ~Q.zero(expr)),
+            Q.even(arg) >> Q.even(expr),
+            Q.odd(arg) >> Q.odd(expr),
+            Q.integer(arg) >> Q.integer(expr),
+            ]
 
 
 ### Add ##
 
 @class_fact_registry.register(Add)
 def _(expr):
-    allargs_positive = AllArgs(x, Q.positive(x))(expr)
-    return Implies(allargs_positive, Q.positive(expr))
-
-@class_fact_registry.register(Add)
-def _(expr):
-    allargs_negative = AllArgs(x, Q.negative(x))(expr)
-    return Implies(allargs_negative, Q.negative(expr))
-
-@class_fact_registry.register(Add)
-def _(expr):
-    allargs_real = AllArgs(x, Q.real(x))(expr)
-    return Implies(allargs_real, Q.real(expr))
-
-@class_fact_registry.register(Add)
-def _(expr):
-    allargs_rational = AllArgs(x, Q.rational(x))(expr)
-    return Implies(allargs_rational, Q.rational(expr))
-
-@class_fact_registry.register(Add)
-def _(expr):
-    allargs_integer = AllArgs(x, Q.integer(x))(expr)
-    return Implies(allargs_integer, Q.integer(expr))
-
-@class_fact_registry.register(Add)
-def _(expr):
-    onearg_notinteger = ExactlyOneArg(x, ~Q.integer(x))(expr)
-    return Implies(onearg_notinteger, ~Q.integer(expr))
+    return [AllArgs(x, Q.positive(x))(expr) >> Q.positive(expr),
```

Comment:
Rather than having `AllArgs` be callable would it not make more sense just to combine this to something like:
```
AllArgs(expr, x, Q.positive(x))
```

Reply:
Done. `AllArgs` and other helpers do not need to be class anymore, so I changed them to function.